app/assets/stylesheets/helpers/_contents-list.scss

```diff
@@ -0,0 +1,91 @@
+.contents-list {
+  margin-bottom: $gutter-two-thirds;
+
+  // Always render the contents list above a
+  // back to contents link
+  position: relative;
+  z-index: 1;
+  background: $white;
+  box-shadow: 0 20px 15px -10px $white;
+
+  li {
+    padding-top: $gutter / 4;
+    line-height: 1.3;
+    list-style-type: none;
+  }
+
+  // Remove underlines from lists of links
+  // to improve legibility
+  a {
+    text-decoration: none;
+
+    &:hover,
+    &:focus,
+    &:active {
+      text-decoration: underline;
+    }
+  }
+
+  &.contents-list-nested,
+  &.contents-list-dashed {
+    li {
+      margin-left: $gutter-half + 5;
+      padding-right: $gutter-half + 5;
+      list-style-type: disc;
+
+      @supports (content: "—") {
+        list-style-type: none;
+
+        &:before {
+          content: "— ";
+          position: relative;
+          float: left;
+          width: $gutter-half + 5;
+          margin-left: -($gutter-half + 5);
+        }
+      }
+    }
+  }
+
+  // When nesting contents lists, the top level items have no
+  // list marker and are styled in bold
+  &.contents-list-nested {
+    .nested-contents-item-h2 {
+      margin-left: 0;
+      list-style-type: none;
+
+      &:before {
+        content: '';
+      }
+    }
+
+    .nested-contents-link-h2 {
+      font-weight: bold;
+    }
+  }
+
+  &.contents-list-numbered {
+    a {
+      display: table;
+    }
+
+    .contents-text,
+    .contents-number {
+      display: table-cell;
+    }
+
+    .contents-number {
+      min-width: 1.5em;
+    }
+
+    .contents-text {
+      $contents-text-padding: 0.3em;
+      padding-left: $contents-text-padding;
+
+      .direction-rtl & {
```

Comment:
This is the same code as was tested successfully in #395, just moved to a new file.